### PR TITLE
Improved nodes and +3 new nodes

### PR DESCRIPTION
**Pull request by knowledgenude**
Now i think this PR makes more sense

* Improved `Get Visible` and `Set Visible` nodes.
* Added `Is Rigid Body`, `Get Object Offscreen` and `Get Rigid Body Data` nodes.
* Temporary fix for: https://github.com/armory3d/armory/issues/1830
* Implemented Children checkbox in `Set Visible` node thanks to @QuantumCoderQC [here](https://forums.armory3d.org/t/help-with-arrays/4207/5) / Closes: https://github.com/armory3d/armory/issues/1558

Preview: 
screenshot1

Test file: [visible_test.zip](https://github.com/armory3d/armory/files/5202475/visible_test.zip)

### CODE REVIEW

File: Sources/armory/logicnode/ClearParentNode.hx [hunk 1]
```diff
@@ -19,9 +19,13 @@ class ClearParentNode extends LogicNode {
 
 		#if arm_physics
 		var rigidBody = object.getTrait(RigidBody);
-		if (rigidBody != null) rigidBody.syncTransform();
+
+		if (rigidBody != null) {
+		rigidBody.animated = false; // Workaround
```
Comment (QuantumCoderQC) [inline Sources/armory/logicnode/ClearParentNode.hx:24]: @knowledgenude , Thank you for the fixes,
I do not fully understand the use of this line here. What is its purpose? 
Setting `rigidBody.animated = false` only affects the trait and not the actual bullet rigid body.

**Comment by knowledgenude** [comment]
@QuantumCoderQC , this is not a correct fix, but it is the only way i found to get `Set Parent` working with rigid bodies. Note that i Set the animated setting trough `Set Parent` and in `Clear Parent` i have to disable animated or the rigid body will be not "active".

File: Sources/armory/logicnode/ClearParentNode.hx (hunk 1 above)
Comment (knowledgenude) [reply Sources/armory/logicnode/ClearParentNode.hx:24]: @QuantumCoderQC , this is not a correct fix, but it is the only way i found to get Set Parent working with rigid bodies. Note that i Set the animated setting trough Set Parent and in Clear Parent i have to disable animated or the rigid body will be not "active".

**Comment by QuantumCoderQC** [comment]
@knowledgenude Okay. Now I understand the purpose.

But if I may, in general, the RigidBodies should not, and cannot be a child object. If such a situation arises where parenting is necessary, the RigidBody trait must be removed, and then after clearing the parent, the trait is added back. This is my opinion.

**Comment by knowledgenude** [comment]
@QuantumCoderQC I also have this opinion, but how is the best way to do that? Because if i remove the rigid body with Set Parent, i need to store its settings in same way that can be applied bacl in Clear Parent. Is possible to simple disable its collision and activation? Activation yes, but even collision mask is not possible currently

**Comment by QuantumCoderQC** [comment]
@knowledgenude  

> Edit: for me i would simply remove the rigid body reference from both nodes and let the users to remove the rigid body themselves and re-add in the moment they want.

Yes. This is better. I am already working on nodes to dynamically add and remove rigid bodies. If I get some time, I will make it happen and create a pull request. Then we can have a discussion about the workflow for parenting rigid bodies. 

Let me know if you agree with this.

**Comment by knowledgenude** [comment]
@QuantumCoderQC , i totally agree :+1: 

I will keep this PR as it is for and i will fix it when you comes with the solution.

**Pull request closed without merging**